Grandpax/Grandpax/SwiftUI/SessionsView+Model/SessionsViewModel.swift

```
@@ -35,6 +35,16 @@ class SessionsViewModel: ObservableObject {
         }
     }
     
+    func presentActivityViewControllerWithSnapshot(_ snapshot: String?) -> ActivityViewController {
+        if let snapshot {
+            let imageData = Data(base64Encoded: snapshot) ?? Data()
+            let items: [AnyObject] = [UIImage(data: imageData) ?? UIImage()]
+            return ActivityViewController(activityItems: items)
+        }
+        
+        return ActivityViewController(activityItems: [])
```

Maybe throw an error if no map snapshot could be found